Sources/SwifterSwift/SwiftStdlib/ArrayExtensions.swift

```diff
@@ -131,3 +131,50 @@ public extension Array where Element: Equatable {
         return filter { set.insert($0[keyPath: path]).inserted }
     }
 }
+
+public extension Array {
+    /// The inheritance system is UIButton -> UIControl -> UIView
+    /// Both U type and U subclasses will be matched
+    ///
+    ///     [UIButton(), UIControl(),UIView()]
+    ///         .filter(kindOf: UIControl.self) ->
+    ///     [<UIButton: 0x7fa710cf54c0>, <UIControl: 0x7fa671429950>]
+    /// - Parameter : A data type
+    /// - Returns: A set of the elements that isIncluded allows.
+    func filter<U>(kindOf _: U.Type) -> [U] {
```

Comment:
Can you elaborate on why this wrapper would be useful? Because to me this would be just abstract some operation that is already trivial e.g.
```
[UIButton(), UIControl(), UIView()].compactMap { $0 as? UIControl }  //   [UIControl]
```
So this may look a bit over abstract something that is already simple.